includes/wikia/services/HubService.class.php

```diff
@@ -61,6 +61,41 @@ public static function getCurrentWikiaVerticalName() {
 	}
 
 	/**
+	 * Get canonical vertical name for given cityId.
+	 * For Lifestyle and Gaming return their names.
+	 * For rest of values return Entertainment.
+	 * @param $cityId
+	 * @return Boolean|String
+	 */
+	public static function getVerticalNameForComscore( $cityId ) {
+		$verticalId = WikiFactoryHub::getInstance()->getVerticalId( $cityId );
+
+		switch ($verticalId) {
+			case WikiFactoryHub::HUB_ID_VIDEO_GAMES:
+				return 'gaming';
+			case WikiFactoryHub::HUB_ID_LIFESTYLE:
+				return 'lifestyle';
+			case WikiFactoryHub::HUB_ID_OTHER:
+				if ( self::isWikiCorporate( $cityId ) ) {
+					return false;
+				} else {
+					return 'lifestyle';
+				}
+			default:
+				return 'entertainment';
+		}
+	}
+
+	private static function isWikiCorporate( $cityId ) {
```

Comment:
Having a generic utility function that can get the vertical name for any wiki is useful.  Perhaps that could be moved to the WikiFactoryHub class?  

It looks like this function is always called in the context of the current wiki (wgCityId) so you could also check the wgEnableWikiaHomePageExt variable directly instead of doing a query here, right?